### PR TITLE
Dedicated quote websocket

### PR DESCRIPTION
**Pull request by da-kami**
After running into too many issues with `js-libp2p` + ws + Tor we decided to add a dedicated websockets for quotes.

### CODE REVIEW

**Comment by da-kami** [comment]
note: ran into binary upload issue with GitHub on previous CI run.

**Review by bonomat — APPROVED** [review]
Awesome seeing this working and thanks for taking over. Only concern is the error handling.

File: swap/src/asb/quote_websocket.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            Ok(latest_price_update) => match Rate::new(latest_price_update.ask, spread).ask() {
+                Ok(amount) => WebsocketBidQuote::Quote(BidQuote {
+                    price: amount,
+                    min_quantity: min,
+                    max_quantity: max,
+                }),
+                Err(e) => {
+                    tracing::error!("Failed to create quote for quote websocket: {:#}", e);
+                    WebsocketBidQuote::Error(Error::BidQuoteError)
+                }
```
[the remark this answers is not in the document]
Comment (bonomat) [reply swap/src/asb/quote_websocket.rs:57]: Found it: we should return a different Message to the client and send an error code from here: 

https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent


https://github.com/seanmonstar/warp/blob/46734106f6a2d523c0dd6b5c6a8415350acb5b72/src/filters/ws.rs#L313-L320

File: swap/src/network/transport.rs [hunk 2]
```diff
@@ -23,10 +21,8 @@ pub fn build_clear_net(id_keys: &identity::Keypair) -> Result<SwapTransport> {
 
     let tcp = TokioTcpConfig::new().nodelay(true);
     let dns = TokioDnsConfig::system(tcp)?;
-    let websocket = WsConfig::new(dns.clone());
 
-    let transport = websocket
-        .or_transport(dns)
+    let transport = dns
```
Comment (bonomat) [inline swap/src/network/transport.rs:25]: Note: you will need to update the config wizard and remove the setting which says we are listening for ` /ip4/0.0.0.0/tcp/9940/ws`

**Review by thomaseizinger — COMMENTED** [review]
Before we introduce this, can we have a detailed technical write-up of what exactly the issues are in why websockets using libp2p don't work?

Using the same underlying technology (websockets) with a different messaging format (hardrolled vs. libp2p) is unlikely to solve any fundamental incompatibility problems, hence I think it is really important for us to know, where the limitations are so we can fix them and eventually remove tech debt like this again.

**Comment by cimble-code** [comment]
I just tried to compile and run this branch on `Windows 10 64 Bit` but it fails to setup the websocket listener.

```
.\asb.exe --testnet start
←[2m2021-06-13 12:26:10←[0m ←[32m INFO←[0m Initialized tracing level=debug
←[2m2021-06-13 12:26:10←[0m ←[32m INFO←[0m Using config file at path=C:\Users\dev2\AppData\Roaming\xmr-btc-swap\config\asb\testnet\config.toml
←[2m2021-06-13 12:26:10←[0m ←[32m INFO←[0m Database and Seed will be stored in db_folder=C:\Users\dev2\AppData\Roaming\xmr-btc-swap\data\asb\testnet
←[2m2021-06-13 12:26:10←[0m ←[34mDEBUG←[0m Opening database at C:\Users\dev2\AppData\Roaming\xmr-btc-swap\data\asb\testnet\database
←[2m2021-06-13 12:26:10←[0m ←[34mDEBUG←[0m Reading in seed from C:\Users\dev2\AppData\Roaming\xmr-btc-swap\data\asb\testnet\seed.pem
←[2m2021-06-13 12:26:10←[0m ←[34mDEBUG←[0m Opening Bitcoin wallet
←[2m2021-06-13 12:26:30←[0m ←[34mDEBUG←[0m Opening Monero wallet
←[2m2021-06-13 12:26:33←[0m ←[34mDEBUG←[0m Opened Monero wallet monero_wallet_name=asb-wallet
←[2m2021-06-13 12:26:33←[0m ←[32m INFO←[0m Initialized Bitcoin wallet bitcoin_balance=0.00000000 BTC
←[2m2021-06-13 12:26:33←[0m ←[33m WARN←[0m The Monero balance is 0, make sure to deposit funds at monero_address=53kpbD35EQT1NrDzcjECtG4SpCEX3pMj7DmzVcNh3CPc89FpPQbK2TBVJRWNRShYEG6npBHy1y5fRNGXTdzJva4LKBsK15z
←[2m2021-06-13 12:26:35←[0m ←[33m WARN←[0m Tor not found. Running on clear net
←[2m2021-06-13 12:26:35←[0m ←[32m INFO←[0m All connections will go through clear net
←[2m2021-06-13 12:26:35←[0m ←[34mDEBUG←[0m Our peer-id peer_id=12D3KooWBahvnaP6iSGHjY7RPKCXnWr3uzAnjK2QqvQAmKgidqjy
Error: Failed to listen on network interface /ip4/0.0.0.0/tcp/9940/ws

Caused by:
    Multiaddr is not supported: /ip4/0.0.0.0/tcp/9940/ws
```

**Comment by thomaseizinger** [comment]
It is my understanding that we don't actually need this now? @da-kami made a PoC of how to access the current quote protocol using `js-libp2p`: https://github.com/da-kami/js-libp2p-quote-poc

@da-kami Please re-open if I misunderstood that!

**Pull request closed without merging**